### PR TITLE
Add contract boilerplate macro

### PR DESCRIPTION
**Pull request by david-yan**
Removes unnecessary boilerplate code from Rust contracts.

### CODE REVIEW

File: src/macros.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//! Macros for generating boilerplate Rust smart contract code.
```
Comment (armaniferrante) [inline src/macros.rs:1]: Lets put 1) an example of the macro being used here and 2) explanations for what deploy and call do and why we use this macro.

File: src/macros.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		pub fn call() {
+			let mut endpoint = $endpoint::new($contract{});
+			owasm_ethereum::ret(&endpoint.dispatch(&owasm_ethereum::input()));
+		}
+	}
+}
+
+/// Generate all boilerplate code
+#[macro_export]
+macro_rules! contract_boilerplate {
```
Comment (armaniferrante) [inline src/macros.rs:29]: Let's pick a name better than `contract_boilerplate`.

`contract!` is probably better or something else

**Comment by nhynes** [comment]
[An alternative proposal, as requested](https://github.com/oasislabs/rust-contract-tutorial/blob/e2ed6285168156e565cd582a48229f992b6400f9/contracts/step-2/src/lib.rs)

**Comment by david-yan** [comment]
Fixed from https://github.com/oasislabs/owasm-abi/pull/4

**Pull request closed without merging**